packages/flutter/lib/src/gestures/monodrag.dart

```diff
@@ -385,12 +391,170 @@ abstract class DragGestureRecognizer extends OneSequenceGestureRecognizer {
     switch (multitouchDragStrategy) {
       case MultitouchDragStrategy.sumAllPointers:
         result = true;
+      case MultitouchDragStrategy.maxAllPointers:
+        if (_getPrimaryDragAxis() != null) {
+          result = true;
+        } else {
+          // Keep the PanGestureRecognizer to latestPointer behavior in this case.
+          result = _acceptedActivePointers.length <= 1 || pointer == _acceptedActivePointers.last;
+        }
       case MultitouchDragStrategy.latestPointer:
         result = _acceptedActivePointers.length <= 1 || pointer == _acceptedActivePointers.last;
     }
     return result;
   }
 
+  void _recordMoveDeltaIfNeeded(int pointer, Offset localDelta) {
+    if (multitouchDragStrategy != MultitouchDragStrategy.maxAllPointers) {
+      assert(_frameTimeStamp == null);
+      assert(_moveDeltasBetweenFrames.isEmpty);
+      return;
+    }
+
+    final Duration currentSystemFrameTimeStamp = SchedulerBinding.instance.currentSystemFrameTimeStamp;
```

Comment:
Fortunately, we can use this to identify whether the events are of the same batch. CC @dkwingsmt

Reply:
Brilliant!